PhysicsTools/PatAlgos/python/slimming/applySubstructure_cfi.py

```diff
@@ -0,0 +1,158 @@
+
+def applySubstructure( process ) :
+
+    from PhysicsTools.PatAlgos.tools.jetTools import addJetCollection
+
+    # add CMS top tagger
+    from RecoJets.JetProducers.caTopTaggers_cff import caTopTagInfos
+    process.caTopTagInfos = caTopTagInfos.clone()
+    process.caTopTagInfosPAT = cms.EDProducer("RecoJetDeltaRTagInfoValueMapProducer",
+                                    src = cms.InputTag("ak8PFJetsCHS"),
+                                    matched = cms.InputTag("cmsTopTagPFJetsCHS"),
+                                    matchedTagInfos = cms.InputTag("caTopTagInfos"),
+                                    distMax = cms.double(0.8)
+    )    
+    
+    #add AK8
+    addJetCollection(process, labelName = 'AK8',
+                     jetSource = cms.InputTag('ak8PFJetsCHS'),
+                     algo= 'AK', rParam = 0.8,
+                     jetCorrections = ('AK8PFchs', cms.vstring(['L1FastJet', 'L2Relative', 'L3Absolute']), 'None'),
+                     btagInfos = ['caTopTagInfosPAT'],
+                     genJetCollection = cms.InputTag('slimmedGenJetsAK8')
+                     )
+    process.patJetsAK8.userData.userFloats.src = [] # start with empty list of user floats
+    process.selectedPatJetsAK8.cut = cms.string("pt > 50")
+
+
+
+    ## AK8 groomed masses
+    from RecoJets.Configuration.RecoPFJets_cff import ak8PFJetsCHSPruned, ak8PFJetsCHSSoftDrop, ak8PFJetsCHSFiltered, ak8PFJetsCHSTrimmed 
+    process.ak8PFJetsCHSPruned   = ak8PFJetsCHSPruned.clone()
+    process.ak8PFJetsCHSSoftDrop = ak8PFJetsCHSSoftDrop.clone()
+    process.ak8PFJetsCHSTrimmed  = ak8PFJetsCHSTrimmed.clone()
+    process.ak8PFJetsCHSFiltered = ak8PFJetsCHSFiltered.clone()
+    process.load("RecoJets.JetProducers.ak8PFJetsCHS_groomingValueMaps_cfi")
+    process.patJetsAK8.userData.userFloats.src += ['ak8PFJetsCHSPrunedMass','ak8PFJetsCHSSoftDropMass','ak8PFJetsCHSTrimmedMass','ak8PFJetsCHSFilteredMass']
+
+    # Add AK8 top tagging variables
+    process.patJetsAK8.tagInfoSources = cms.VInputTag(cms.InputTag("caTopTagInfosPAT"))
+    process.patJetsAK8.addTagInfos = cms.bool(True)
+
+
+
+    # add Njetiness
+    process.load('RecoJets.JetProducers.nJettinessAdder_cfi')
+    process.NjettinessAK8 = process.Njettiness.clone()
+    process.NjettinessAK8.src = cms.InputTag("ak8PFJetsCHS")
+    process.NjettinessAK8.cone = cms.double(0.8)
+    process.patJetsAK8.userData.userFloats.src += ['NjettinessAK8:tau1','NjettinessAK8:tau2','NjettinessAK8:tau3']
+
+
+    ## PATify pruned fat jets
+    addJetCollection(
+        process,
+        labelName = 'AK8PFCHSSoftDrop',
+        jetSource = cms.InputTag('ak8PFJetsCHSSoftDrop'),
+        btagDiscriminators = ['None'],
+        jetCorrections = ('AK8PFchs', ['L1FastJet', 'L2Relative', 'L3Absolute'], 'None'),
+        getJetMCFlavour = False # jet flavor disabled
+    )
+    getattr(process,'patJetPartonMatchAK8PFCHSSoftDrop').matched = cms.InputTag('prunedGenParticles')
```

Comment:
Shouldn't then all PAT jets be switched to `prunedGenParticles` for jet-parton matching? I guess `mcStatus` should be updated as well.